lib/perl/Genome/Model/Tools/Bwa/RunMem.pm

```diff
@@ -403,7 +403,10 @@ sub _sort_commands {
     my $comp_level = 0;
     my $threads = $self->num_threads;
     my $per_thread_mem = $self->_sort_memory_per_thread_mb;
-    my $tmp_path = File::Spec->catfile($self->temp_dir, "samtools-sort");
+    my $tmp_path = $self->temp_dir;
+
+    $tmp_path = Genome::Sys->create_temp_file_path unless $tmp_path;
```

Comment:
+1, given the priority.  Many X Ten validation alignments are failing because of this.  Can we merge ASAP unless there are objections?

My assumption is that one can define temp_dir with a network disk location or a defined output location to monitor the temp usage.  I'm also assuming that in our pipelines no temp_dir is defined so `Genome::Sys->create_temp_file_path` turns out to be the method called.